### ISSUE TITLE
Identifying Response Content as List using TypeHint

### ISSUE THREAD

**Issue by sg-ad** [stoicflame/enunciate#230]
Hi,

Suppose that my API returns a collection of addresses wrapped in a `Response`.  The best I can do is to indicate the type of the actual objects in the collection, but is there a way to show that it's actually a collection of these objects that will be returned?  If not, it would be a great enhancement!

For example, the following code
```java
@GET
@Path("myAddresses")
List<AddressWS> readAddresses();
```
yields
```
application/json array of AddressWS (JSON)
application/xml list of addressWS (XML)
```
Now if my code is like this
```java
@GET
@Path("user/{userId}/address")
@TypeHint(AddressWS.class)
Response readUserAddresses(@PathParam("userId") String userId) {...}
```
I get
```
application/json AddressWS (JSON)
application/xml addressWS (XML)
```
Ideally there would be a way for me to get `array of AddressWS` instead of just `AddressWS`.

BTW, is the first character of the XML object lowercase on purpose?

Thanks!

**Comment by stoicflame** [comment]
Tracking as an enhancement.

The XML object is lower-case as dictated by JAXB for the name XML type. To customize the name of the XML type, you can use the `@XmlType` annotation.

**Comment by sg-ad** [comment]
If you want, I'd be happy to take a stab at implementing this enhancement.  I think that collections are the only untyped wrappers that are of concern.  Would simply adding a new attribute to the `@TypeHint` annotation suffice?  Something like `asCollectionOf` with default to `false`?

**Comment by stoicflame** [comment]
Sure, I'd be happy to take a pull request.

Your proposed solution sounds good to me.

**Comment by stoicflame** [comment]
Fixed at 91767fc.

**Comment by stoicflame** [comment]
Okay, I botched this one. 91767fc has been reverted at f791989 because there was already a way to specify a list of things using `@TypeHint`: just use an array declaration, e.g.:

```java
@TypeHint(AddressWS[].class)
```
